### ISSUE TITLE
python drivers support db-url schemas

### ISSUE THREAD

**Issue by digi604** [rethinkdb/rethinkdb#4101]
Please support dburl schemas for connections.

[protocol]://[node]/[database]?[options]

this would be something like:

rethinkdb://myhost:1234/myDB

**Comment by danielmewes** [comment]
Since this is pretty common in other DBs, I think it would be useful (it could be a separate `connect` optarg `url: ...`).

I cannot promise how quickly we'll get to this, but we'll keep it on our list.

Also marking as good_for_contributors.

**Comment by sidthekidder** [comment]
Hey, I'd like to start contributing to RethinkDB and this seems a good first issue. I would basically have to modify the init function of CommonOptionsParser in [utils_common.py](https://github.com/rethinkdb/rethinkdb/blob/next/drivers/python/rethinkdb/utils_common.py) right, to check for a `url` opt?

**Comment by adriantodt** [comment]
Implemented it on the Java driver, on the `develop/experimental` branch.

**Comment by gabor-boros** [comment]
Also, we should support all the connection options through the URL except SSL, so:
host, port, db, auth_key, user, password, timeout

In this case a connection string would look like rethinkdb://user:password@myhost:1234/myDB?auth_key=mykey&timeout=20 where user, password, auth_key and timeout are optional but the other is required

**Comment by adriantodt** [comment]
Also, I'd suggest adding the db-url syntax to docs, so the driver's documentation can just point to the page on the docs.

**Comment by gabor-boros** [comment]
@adriantodt I agree to add the syntax to the docs. The only question is when do we want to release that "feature"/syntax change to the users. I mean ship it with the 2.5.0 release or before? Also, if we extend the docs and add the syntax change to the drivers' API documentation, what do you mean by "driver's documentation can just point to the page on the docs"?

**Comment by adriantodt** [comment]
> The only question is when do we want to release that "feature"/syntax change to the users. I mean ship it with the 2.5.0 release or before?

There's just so much stuff going on that there's already enough to do a 2.4.1 release for Java and Python drivers, and maybe minor server stuff as well?

> what do you mean by "driver's documentation can just point to the page on the docs"?

Java documentation is just sugared HTML, I can point the users that use `db-url` with a http link into the site documentation.

**Comment by gabor-boros** [comment]
As a side note, the python driver is at 2.4.4, because the idea there was to keep 2.4 but the patch version. So 2.4.5 for python, and 2.4.1 for ruby, JavaScript and Java could work I think
